src/switch_ivr_originate.c

```diff
@@ -3293,6 +3309,14 @@ SWITCH_DECLARE(switch_status_t) switch_ivr_originate(switch_core_session_t *sess
 					switch_ivr_parse_all_events(oglobals.session);
 				}
 
+				if (global_elapsed > (time_t) request_timelimit_sec) {
+					oglobals.idx = IDX_TIMEOUT;
+					if (force_reason == SWITCH_CAUSE_NONE) {
+						force_reason = SWITCH_CAUSE_PROGRESS_TIMEOUT;
```

Comment:
Could we introduce new force_reason something like `SWITH_CAUSE_GPROGRESS_TIMEOUT`?
May we will need to distinguish which time out caused call hangup in the future..

Reply:
Hm.. I think adding a new variable is out of this scope. I think this should be OK for now.